core/src/collections.rs

```diff
@@ -1,10 +1,194 @@
-use crate::conn::MongoConnection;
-use crate::err::Result;
-use crate::stmt::MongoStatement;
+use crate::{
+    col_metadata::{ColumnNullability, MongoColMetadata},
+    conn::MongoConnection,
+    err::Result,
+    json_schema::{
+        simplified::{Atomic, Schema},
+        BsonTypeName,
+    },
+    stmt::MongoStatement,
+};
 use bson::Bson;
+use lazy_static::lazy_static;
 use mongodb::results::CollectionSpecification;
 use mongodb::sync::Cursor;
 
+lazy_static! {
+    static ref COLLECTIONS_METADATA: Vec<MongoColMetadata> = vec![
+        MongoColMetadata::new(
+            "",
+            "".to_string(),
+            "TABLE_CAT".to_string(),
+            Schema::Atomic(Atomic::Scalar(BsonTypeName::String)),
+            ColumnNullability::NoNulls
+        ),
+        MongoColMetadata::new(
+            "",
+            "".to_string(),
+            "TABLE_SCHEM".to_string(),
+            Schema::Atomic(Atomic::Scalar(BsonTypeName::String)),
+            ColumnNullability::Nullable
+        ),
+        MongoColMetadata::new(
+            "",
+            "".to_string(),
+            "TABLE_NAME".to_string(),
+            Schema::Atomic(Atomic::Scalar(BsonTypeName::String)),
+            ColumnNullability::NoNulls
+        ),
+        MongoColMetadata::new(
+            "",
+            "".to_string(),
+            "TABLE_TYPE".to_string(),
+            Schema::Atomic(Atomic::Scalar(BsonTypeName::String)),
+            ColumnNullability::NoNulls
+        ),
+        MongoColMetadata::new(
+            "",
+            "".to_string(),
+            "REMARKS".to_string(),
+            Schema::Atomic(Atomic::Scalar(BsonTypeName::String)),
+            ColumnNullability::Nullable
+        ),
+    ];
+}
+
+mod unit {
+    #[test]
+    fn metadata_size() {
+        use crate::{collections::MongoCollections, stmt::MongoStatement};
+        assert_eq!(5, MongoCollections::empty().get_resultset_metadata().len());
+    }
+
+    #[test]
+    fn metadata_column_names() {
+        use crate::{collections::MongoCollections, stmt::MongoStatement};
+        // These were generated straight from the docs (hence the - 1). This
+        // gives us assurance that the column names are all correct.
```

Comment:
```suggestion
        // This gives us assurance that the column names are all correct.
```
Dead comment since we got rid of the `- 1`